### PR TITLE
Fix `Actions.escapeLabel` collision if only the repo name differs

### PR DESCRIPTION
**Pull request by fmeum**
This fixes action conflicts when `cc_toolchain`s in different repos, but with identical package and target names, use `dynamic_runtime_lib`.

### CODE REVIEW

**Comment by fmeum** [comment]
@bazel-io flag

**Comment by keertk** [comment]
@bazel-io fork 7.3.0

File: src/main/java/com/google/devtools/build/lib/actions/Actions.java [hunk 1]
```diff
@@ -371,7 +373,19 @@ public static String escapedPath(String path) {
    * that no other label maps to this string.
    */
   public static String escapeLabel(Label label) {
-    return PATH_ESCAPER.escape(label.getPackageName() + ":" + label.getName());
+    String path = label.getPackageName() + ":" + label.getName();
+    if (!label.getRepository().isMain()) {
```
Comment (comius) [inline src/main/java/com/google/devtools/build/lib/actions/Actions.java:377]: Are there any alternative solutions?

I know of an attempt by TF team to demangle the library in @rules_cc/cc/private/toolchain/osx_cc_wrapper.sh.tpl

Using one way hash would prevent that.

Comment (fmeum) [reply src/main/java/com/google/devtools/build/lib/actions/Actions.java:377]: We can use the name as is and wait for but reports that may never come 🙂

Just to understand this better, why would they need to demangle the name?

Comment (comius) [reply src/main/java/com/google/devtools/build/lib/actions/Actions.java:377]: I think it’s to compile on RBE and run locally.

Comment (fmeum) [reply src/main/java/com/google/devtools/build/lib/actions/Actions.java:377]: I implemented a reversible scheme, PTAL.

**Review by comius — APPROVED** [review]

**Pull request closed without merging**